app/src/main/java/com/google/android/gms/samples/wallet/PaymentsUtil.kt

```diff
@@ -0,0 +1,272 @@
+        @Throws(JSONException::class)
+        get() = JSONObject().put("merchantName", "Example Merchant")
+
+    /**
+     * Creates an instance of [PaymentsClient] for use in an [Activity] using the
+     * environment and theme set in [Constants].
+     *
+     * @param activity is the caller's activity.
+     */
+    fun createPaymentsClient(activity: Activity): PaymentsClient {
+        val walletOptions = Wallet.WalletOptions.Builder().setEnvironment(Constants.PAYMENTS_ENVIRONMENT).build()
+        return Wallet.getPaymentsClient(activity, walletOptions)
+    }
+
+    /**
+     * Provide Google Pay API with a payment amount, currency, and amount status.
+     *
+     * @return information about the requested payment.
+     * @throws JSONException
+     * @see [TransactionInfo](https://developers.google.com/pay/api/android/reference/object.TransactionInfo)
+     */
+    @Throws(JSONException::class)
+    private fun getTransactionInfo(price: String): JSONObject {
+        val transactionInfo = JSONObject()
+        transactionInfo.put("totalPrice", price)
+        transactionInfo.put("totalPriceStatus", "FINAL")
+        transactionInfo.put("currencyCode", Constants.CURRENCY_CODE)
+
+        return transactionInfo
+    }
+
+    /**
+     * An object describing information requested in a Google Pay payment sheet
+     *
+     * @return Payment data expected by your app.
+     * @see [PaymentDataRequest](https://developers.google.com/pay/api/android/reference/object.PaymentDataRequest)
+     */
+    fun getPaymentDataRequest(price: String): Optional<JSONObject> {
+        try {
+            val paymentDataRequest = baseRequest
+            paymentDataRequest.put(
+                    "allowedPaymentMethods", JSONArray().put(cardPaymentMethod()))
+            paymentDataRequest.put("transactionInfo", getTransactionInfo(price))
+            paymentDataRequest.put("merchantInfo", merchantInfo)
+
+            /* An optional shipping address requirement is a top-level property of the PaymentDataRequest
+      JSON object. */
+            paymentDataRequest.put("shippingAddressRequired", true)
+
+            val shippingAddressParameters = JSONObject()
+            shippingAddressParameters.put("phoneNumberRequired", false)
+
+            val allowedCountryCodes = JSONArray(Constants.SHIPPING_SUPPORTED_COUNTRIES)
+
+            shippingAddressParameters.put("allowedCountryCodes", allowedCountryCodes)
+            paymentDataRequest.put("shippingAddressParameters", shippingAddressParameters)
+            return Optional.of(paymentDataRequest)
+        } catch (e: JSONException) {
+            return Optional.empty()
+        }
+
+    }
+}
+
+
+/**
+ * Converts micros to a string format accepted by [PaymentsUtil.getPaymentDataRequest].
+ *
+ * @param micros value of the price.
+ */
+fun Long.microsToString(): String {
+    return BigDecimal(this).divide(PaymentsUtil.MICROS).setScale(2, RoundingMode.HALF_EVEN).toString()
```

Comment:
Sounds like a very important function, I'm not checking if it's doing what it's supposed to. I'd add tests for this but I guess the Java version didn't have them.

Reply:
Adding unit tests is out of scope of this PR